tests/test_dev.py

```diff
@@ -54,58 +43,99 @@ def func_a(self):
                 return "a"
 
         with warnings.catch_warnings(record=True) as w:
-            # Cause all warnings to always be triggered.
-            warnings.simplefilter("always")
             # Trigger a warning.
-            assert a().property_b == "b"
-            # Verify some things
+            assert TestClass().property_b == "b"
+            # Verify warning type
             assert issubclass(w[-1].category, FutureWarning)
 
         with warnings.catch_warnings(record=True) as w:
-            # Cause all warnings to always be triggered.
-            warnings.simplefilter("always")
             # Trigger a warning.
-            assert a().func_a() == "a"
+            assert TestClass().func_a() == "a"
             # Verify some things
             assert issubclass(w[-1].category, FutureWarning)
 
     def test_deprecated_classmethod(self):
-        class A:
-            def __init__(self):
-                pass
+        class TestClass:
+            """A dummy class for tests."""
 
             @classmethod
-            def classmethod_a(self):
+            def classmethod_a(cls):
                 pass
 
             @classmethod
             @deprecated(classmethod_a)
-            def classmethod_b(self):
+            def classmethod_b(cls):
                 return "b"
 
         with warnings.catch_warnings(record=True) as w:
-            # Cause all warnings to always be triggered.
-            warnings.simplefilter("always")
             # Trigger a warning.
-            assert A().classmethod_b() == "b"
+            assert TestClass().classmethod_b() == "b"
             # Verify some things
             assert issubclass(w[-1].category, FutureWarning)
 
-        class A:
-            def __init__(self):
-                pass
+        class TestClass_deprecationwarning:
+            """A dummy class for tests."""
 
             @classmethod
-            def classmethod_a(self):
+            def classmethod_a(cls):
                 pass
 
             @classmethod
             @deprecated(classmethod_a, category=DeprecationWarning)
-            def classmethod_b(self):
+            def classmethod_b(cls):
                 return "b"
 
         with pytest.warns(DeprecationWarning):
-            assert A().classmethod_b() == "b"
+            assert TestClass_deprecationwarning().classmethod_b() == "b"
+
+    def test_deprecated_deadline(self):
+        @deprecated(deadline=(2000, 1, 1))
+        def func_old():
+            pass
+
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
let's avoid single-letter variable names and call this `warnings` (likewise below)

Reply:
Sure! Would change that.